main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -824,6 +824,96 @@ object Typer extends Phase[ResolvedAst.Program, TypedAst.Root] {
             resultType <- unifyM(tvar, Type.Unit, loc)
           } yield resultType
 
+        /*
+         * NewChannel expression.
+         */
+        case ResolvedAst.Expression.NewChannel(exp, tpe, loc) =>
+          //
+          //  e: Int8 | Int16 | Int32 | Int64 | BigInt
+          //  ----------------------------------------
+          //  channel t e : Channel[t]
+          //
+          for {
+            texp <- visitExp(exp)
+            _ <- texp match {
+              case Type.Int8 => unifyM(texp, Type.Int8, loc)
+              case Type.Int16 => unifyM(texp, Type.Int16, loc)
+              case Type.Int32 => unifyM(texp, Type.Int32, loc)
+              case Type.Int64 => unifyM(texp, Type.Int64, loc)
+              case Type.BigInt => unifyM(texp, Type.BigInt, loc)
+              case _ => unifyM(texp, Type.Int32, loc)
+            }
+              
+            resultType <- liftM(Type.mkChannel(tpe))
+          } yield resultType
+
+        /*
+         * GetChannel expression.
+         */
+        case ResolvedAst.Expression.GetChannel(exp, tvar, loc) =>
+          // exp : Channel[t]
+          // ----------------
+          // <- exp : t
+          for (
+            tpe  <- visitExp(exp);
+            rtpe <- unifyM(tvar, tpe, loc)
+          ) yield rtpe
+
+        /*
+         * PutChannel expression.
+         */
+        case ResolvedAst.Expression.PutChannel(exp1, exp2, tvar, loc) =>
+          // exp1 : Channel[t]    exp2: t
+          // ----------------------------
+          // exp1 <- exp2 : Channel[t]
+          for (
+            tpe1 <- visitExp(exp1);
+            tpe2 <- visitExp(exp2);
+            rtpe <- unifyM(tvar, tpe1, tpe2, loc)
+          ) yield rtpe
+
+        /**
+          * Spawn expression.
+          */
+        case ResolvedAst.Expression.Spawn(exp, tvar, loc) =>
+          // exp : () -> Unit
+          // ----------------
+          // spawn exp : Unit
+          for (
+            tpe <- visitExp(exp);
+            e <- unifyM(tpe, Type.mkArrow(Type.Unit, Type.Unit), loc);
+            resultType <- unifyM(tvar, Type.Unit, loc)
+          ) yield resultType
+
+        /**
+          * SelectChannel expression.
+          */
+        case ResolvedAst.Expression.SelectChannel(rules, tvar, loc) =>
+          // exp1 : t1   exp2 : Channel[t1]   exp3 : t2
+          // ------------------------------------------ [t-SelectRule]
+          // case exp1 <- exp2 => epx3 : t2
+          //
+          //  sr : t
+          //  ------------------ [t-SelectChannel]
+          //  select { sr+ } : t
+          assert(rules.nonEmpty)
+          // Extract the symbols, channels, and body expressions of each rule-
+          val bodies = rules.map(_.exp)
+
+          val _ = rules map {
```

Comment:
This fragment is incorrect. You *always* have to use the value returned by unify. Not using it means ignoring the result - which makes no sense in a functional language.